### PR TITLE
RSPC Major changes, ready for deployment 

### PR DESCRIPTION
**Pull request by Devendra072002**
### Brief description of what is fixed or changed

- Added Check on the research procedures module that the faculty accessing has the designation 'Dean_RSPC' or not.
- If the logged-in user is not Dean RSPC then the faculty will be redirected to EIS
- Dean RSPC can view all the information about the research activities going on campus
- Dean RSPC can also view his personal profile using the view profile button
- Added user search form, so that faculty can search for information regarding research work going on the campus under particular faculty
- Added Generate citation functionality so that dean rspc can generate citations of projects under given faculty.
- Fixed generate report functionality, earlier the queries were not fetching correct data.
- Integrated faculty information using EIS module models.
## Types of changes

_Put an `x` in the boxes that apply_

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Other (please describe):

## Checklist

_Put an `x` in the boxes that apply_

-   [x] My code follows the [style guidelines of this project](https://docs.google.com/document/d/1GI2Hile8UbGZ82gFe1y4wAVPcNPXip1cmXTzog1S41U/edit)
-   [x] I have performed a self-review of my own code
-   [x] I have created new branch for this pull request
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [ ] My changes does not break the current system and it passes all the current test cases.

### CODE REVIEW

**Comment by ashray0124** [comment]
Approve @akshatnema 
Module name- RSPC (GAD-10)

File: FusionIIIT/applications/eis/api/views.py [hunk 1]
```diff
@@ -87,3 +89,154 @@ def profile(request, username=None):
...
+    #         'consymps' : consymps,
+    #         'awards' : awards,
+    #         'talks' : talks,
+    #         'chairs' : chairs,
+    #         'keynotes' : keynotes,
+    #         'events' : events,
+    #         'year_range' : year_range,
+    #         'faculty_about' : faculty_about
+    # }
+    # return Response(data=resp, status=status.HTTP_200_OK)
```
Comment (akshatnema) [inline FusionIIIT/applications/eis/api/views.py]:
```suggestion
    
```
Please remove unnecessary commented code.

File: FusionIIIT/applications/research_procedures/views.py [hunk 2]
```diff
@@ -230,4 +232,251 @@ def transfer_insert(request):
...
-    return redirect(reverse("research_procedures:patent_registration"))
\ No newline at end of file
+    return redirect(reverse("research_procedures:patent_registration"))
+
+# Dean RSPC Profile
+
+def rspc_profile(request):
+    # user = get_object_or_404(faculty_about, user=request.user)
+    # pf = user.user
+    # all_user = User.objects.all()
```
Comment (akshatnema) [inline FusionIIIT/applications/research_procedures/views.py]:
```suggestion
```
same here

File: FusionIIIT/applications/research_procedures/views.py [hunk 3]
```diff
@@ -230,4 +232,251 @@ def transfer_insert(request):
...
+    # projects = emp_research_projects.objects.all().order_by('-start_date')
+    # consultancy = emp_consultancy_projects.objects.all().order_by('-start_date')
+    # patents = emp_patents.objects.all().order_by('-p_year', '-a_month')
+    # techtransfers = emp_techtransfer.objects.all().order_by('-date_entry')
+    # mtechs = emp_mtechphd_thesis.objects.filter(degree_type=1).order_by('-s_year', '-a_month')
+    # phds = emp_mtechphd_thesis.objects.filter(degree_type=2).order_by('-s_year', '-a_month')
+    # fvisits = emp_visits.objects.filter(v_type=2).order_by('-start_date')
+    # ivisits = emp_visits.objects.filter(v_type=1).order_by('-start_date')
+    # for fvisit in fvisits:
+    #     fvisit.countryfull = countries[fvisit.country]
```
Comment (akshatnema) [inline FusionIIIT/applications/research_procedures/views.py]:
```suggestion
```

**Review by akshatnema — CHANGES_REQUESTED** [review]
@Devendra072002 Kindly remove all the unnecessary commented code from the PR.

**Comment by Devendra072002** [comment]
@akshatnema I have done the changes as requested.

File: FusionIIIT/templates/eisModulenew/rspc_generatereport.html [hunk 4]
```diff
@@ -367,4 +380,23 @@ <h5>TO</h5>
...
+      //     // Send the input data to the Django view using a fetch request
+      //     fetch('http://127.0.0.1:8000/research_procedures/user_profile', {
+      //       method: 'POST',
+      //       body: JSON.stringify({ username }),
+      //       headers: {
+      //         'Content-Type': 'application/json',
+      //         'X-CSRFToken': '{{ csrf_token }}'
+      //       }
+      //     })
+      //   })
```
Comment (akshatnema) [inline FusionIIIT/templates/eisModulenew/rspc_generatereport.html]: Kindly remove this comment also.

File: FusionIIIT/templates/rs/show_projects.html [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+            <div class="fields">
+                <div class="fourteen wide field">
+                    <div class="field">
+                        <label>Title of Project</label>
+                        <input placeholder="Title" id="researchB" type="text" name="title" required>
+                    </div>
+                </div>
+            </div>
+        </div>
+        -->
```
Comment (akshatnema) [inline FusionIIIT/templates/rs/show_projects.html]: Remove this comment too.

**Review by akshatnema — CHANGES_REQUESTED** [review]
Kindly remove all the unnecessary comments you have added to this PR.

**Review by akshatnema — APPROVED** [review]

**Pull request merged**